### PR TITLE
WIP: check on some performance changes

### PR DESCRIPTION
**Pull request by rcritten**
Exercise some changes that should reduce the number of queries performed for some operations, mostly host and user related.

The execution time patches will be replaced by Antonio's patch at some point.

Running in CI to see if anything is terribly broken.

### CODE REVIEW

File: ipaserver/plugins/automount.py [hunk 1]
```diff
@@ -610,8 +611,11 @@ def check_key_uniqueness(self, location, map, **keykw):
         entries = self.methods.find(location, map, automountkey=key)['result']
         if len(entries) > 0:
             if key == DIRECT_MAP_KEY:
+                if 'quiet' in keykw:
+                    newkw = deepcopy(keykw)
```
Comment (tiran) [inline ipaserver/plugins/automount.py:615]: Do you need a deepcopy here? deepcopy recurses into values and makes copies of nested values. I think ``keykw.copy()`` should be good enough

File: contrib/parselog.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import division
```
Comment (tiran) [inline contrib/parselog.py:18]: Only needed for Python 2.

File: ipaserver/plugins/cert.py [hunk 3]
```diff
@@ -1559,6 +1559,12 @@ class cert_find(Search, CertMethod):
             normalizer=normalize_pkidate,
             autofill=False,
         ),
+        StrEnum(
+            'status?',
+            doc=_("Status of the certificate"),
+            values=(u'VALID', u'INVALID', u'REVOKED', u'EXPIRED',
+                    u'REVOKED_EXPIRED'),
+        ),
```
Comment (tiran) [inline ipaserver/plugins/cert.py:1567]: The ``u''`` prefix is no longer required in new code.

**Comment by rcritten** [comment]
Thanks for the review. I'll update if I end up proposing any of these for real.

**Comment by rcritten** [comment]
cert_find changes moved to https://github.com/freeipa/freeipa/pull/5701

**Comment by rcritten** [comment]
Cache ca_is_enabled moved to https://github.com/freeipa/freeipa/pull/5702

**Comment by rcritten** [comment]
Closing this. Most of the fixes were integrated into other PRs or dropped because the general LDAP Cache PR covered it.

**Pull request closed without merging**